content/en/docs/languages/js/propagation.md

```diff
@@ -2,51 +2,257 @@
+```
+
+{{% /tab %}} {{< /tabpane >}}
+
+Start a second shell and run the `client.ts` (or `client.js`):
+
+{{< tabpane text=true >}} {{% tab TypeScript %}}
+
+```shell
+npx ts-node client.ts
+```
+
+{{% /tab %}} {{% tab JavaScript %}}
+
+```shell
+node client.js
+```
+
+{{% /tab %}} {{< /tabpane >}}
+
+Both shells should emit span details to the console. The client output looks
+similar to the following:
+
+```javascript {hl_lines=[7,11]}
+{
+  resource: {
+    attributes: {
+      // ...
+    }
+  },
+  traceId: 'cccd19c3a2d10e589f01bfe2dc896dc2',
+  parentId: undefined,
+  traceState: undefined,
+  name: 'GET',
+  id: '6f64ce484217a7bf',
+  kind: 2,
+  timestamp: 1718875320295000,
+  duration: 19836.833,
+  attributes: {
+    'url.full': 'http://localhost:8080/rolldice',
+    // ...
+  },
+  status: { code: 0 },
+  events: [],
+  links: []
+}
+```
+
+Take note of the traceId (`cccd19c3a2d10e589f01bfe2dc896dc2`) and ID
+(`6f64ce484217a7bf`). Both can be find in the output of client as well:
+
+```javascript {hl_lines=["6-7"]}
+{
+  resource: {
+    attributes: {
+      // ...
+  },
+  traceId: 'cccd19c3a2d10e589f01bfe2dc896dc2',
+  parentId: '6f64ce484217a7bf',
+  traceState: undefined,
+  name: 'GET /rolldice',
+  id: '027c5c8b916d29da',
+  kind: 1,
+  timestamp: 1718875320310000,
+  duration: 3894.792,
+  attributes: {
+    'http.url': 'http://localhost:8080/rolldice',
+    // ...
+  },
+  status: { code: 0 },
+  events: [],
+  links: []
+}
+```
+
+Your client and server application successfully report connected spans. If you
+send both to a backend now the visualization will show this dependency for you.
+
+## Manual context propagation
+
+In some cases, it is not possible to propagate context automatically as outlined
+in previous section. There might not be an instrumentation library that matches
+a library you're using to have services communicate with one another. Or you
+might have requirements that these libraries can't fulfill even if they existed.
 
 When you must propagate context manually, you can use the
 [context API](/docs/languages/js/context).
 
+### Generic example
+
 The following generic example demonstrates how you can propagate trace context
 manually.
 
 First, on the sending service, you'll need to inject the current `context`:
 
+{{< tabpane text=true >}} {{% tab TypeScript %}}
+
+```typescript
+// Sending service
+import { Context, propagation, trace } from '@opentelemetry/api';
+
+// Define an interface for the output object that will hold the trace information.
+interface Carrier {
+  traceparent?: string;
+  tracestate?: string;
+}
+
+// Create an output object that conforms to that interface.
+const output: Carrier = {};
+
+// Serialize the traceparent and tracestate from context into
+// an output object.
+//
+// This example uses the active trace context, but you can
+// use whatever context is appropriate to your scenario.
+propagation.inject(trace.getTracerProvider().getActiveContext(), output);
```

Comment:
Could this use the same `propagation.inject(context.active(), output);` that is used in the JS tab below?

Reply:
Yes 🙌 

```suggestion
propagation.inject(context.active(), output);
```

like this?

Reply:
That's what I meant yes. I haven't tested it myself to be sure.

Actually... just browsing `@opentelemetry/api` code, I don't believe there *is* a `.getActiveContext()` method. I'm curious where that code came from.  It should be `context.active()` -- per https://opentelemetry.io/docs/languages/js/context/#get-active-context